lib/class-core.php

```diff
@@ -158,17 +165,37 @@ public static function get_sites_for_post( $post = null ) {
 	 * @return bool True if yes, false if no.
 	 */
 	public static function is_post_allowed_on_current_site( $post = null ) {
-		$sites = self::get_sites_for_post( $post );
+		return self::is_post_allowed_on_site(
+			$post,
+			intval( self::instance()->get_current_site_term( 'term_id' ) )
+		);
+	}
 
-		// First check if the current site is in the list and prefer that.
-		$current_site_id = intval( self::instance()->get_current_site_term( 'term_id' ) );
-		foreach ( $sites as $site ) {
-			if ( intval( $site->term_id ) === $current_site_id ) {
-				return true;
-			}
+	/**
+	 * Is the given post allowed on the given site?
+	 *
+	 * @param  int|\WP_Post  $post    Post ID or object.
+	 * @param  int           $site_id Site ID.
+	 * @return boolean true if yes, false if no.
+	 */
+	public static function is_post_allowed_on_site( $post, $site_id ) {
+		$post = get_post( $post );
+		if ( is_object_in_taxonomy( $post->post_type, Site::instance()->name ) ) {
```

Comment:
You might want to check that `$post` is of type `\WP_Post` before trying to use `post_type`.